canonicalwebteam/discourse_docs/parsers.py

```diff
@@ -85,21 +106,71 @@ def parse_topic(topic):
     }
 
 
-def parse_navigation(index_soup):
+def parse_navigation(index_soup, url_map):
     """
     Given the HTML soup of a index topic
-    extract the "navigation" section
+    extract the "navigation" section, and rewrite any
+    links in the url_map
     """
 
-    nav_soup = get_section(index_soup, "Content")
+    nav_soup = get_section(index_soup, "Navigation")
     nav_html = "Navigation missing"
 
     if nav_soup:
+        # Convert links to the form needed in this site
+        for link in nav_soup.find_all("a"):
+            if "href" in link.attrs:
+                url = link.attrs["href"]
+                link_match = TOPIC_URL_MATCH.match(url)
+
+                if link_match:
+                    topic_id = int(link_match.groupdict()["topic_id"])
+                    if topic_id in url_map:
+                        link.attrs["href"] = url_map[topic_id]
```

Comment:
Can we be sure that the `topic_id` will be in `url_map`. Does it get instantiated will all links on startup? 
Or should this be added dynamically if a link gets found but isnt mapped yet?

Reply:
We can definitely be sure if `if topic_id in url_map:` is true...

Reply:
Oh yeah, I was thinking about the `else` case. There is no fallback if the link is not found in the map

Reply:
Yes, that's the point. Only if the topic is in the map do we need to do the replacement. Otherwise we leave it as it is.